Lib/shlex.py

```diff
@@ -320,17 +320,14 @@ def join(split_command):
 
 _find_unsafe = re.compile(r'[^\w@%+=:,./-]', re.ASCII).search
 
-def quote(s):
+def quote(s, always=False):
     """Return a shell-escaped version of the string *s*."""
-    if not s:
-        return "''"
-    if _find_unsafe(s) is None:
-        return s
-
-    # use single quotes, and put single quotes into double quotes
-    # the string $'b is then quoted as '$'"'"'b'
-    return "'" + s.replace("'", "'\"'\"'") + "'"
+    if always or not s or _find_unsafe(s) is not None:
```

Comment:
I would prefer to keep the special code path for the empty string (`not s`), with just `return "''"`.

Reply:
For clarity's sake I agree on re-reading.

The reason for the 'simplification' in 5be56a0 was to avoid having a second `"''"` string as well as the `"'" + s.replace("'", "'\"'\"'") + "'"`, a sort of single-source-of-truth / single source of quoting. :smile: 

But really the empty string *still is* a special case (because it requires the `not s` check). All that's happened is the boolean logic on line 325 has been made more complex. Also skipping the `.replace` for the empty string may save a few microseconds lol.

Thus I will revert the special case.